opal/mca/btl/tcp/btl_tcp_proc.c

```diff
@@ -452,6 +452,11 @@ int mca_btl_tcp_proc_insert( mca_btl_tcp_proc_t* btl_proc,
     max_peer_interfaces = proc_data->max_local_interfaces;
     peer_interfaces = (mca_btl_tcp_interface_t**)calloc( max_peer_interfaces, sizeof(mca_btl_tcp_interface_t*) );
     assert(NULL != peer_interfaces);
+    if (NULL == peer_interfaces) {
```

Comment:
shouldn't you remove the `assert()` ?

Reply:
I thought about removing the `assert()` and at the time, decided against it.  But it's probably 60/40 better to remove it and just rely on the returning of `OPAL_ERR_OUT_OF_RESOURCE`.  I'll update.